<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>860</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BÃ©zier Surfaces</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="CPWidget" name="topViewWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>400</width>
      <height>400</height>
     </rect>
    </property>
   </widget>
   <widget class="CPWidget" name="sideViewWidget">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>10</y>
      <width>400</width>
      <height>400</height>
     </rect>
    </property>
   </widget>
   <widget class="CPWidget" name="frontViewWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>400</width>
      <height>400</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>420</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>600</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>390</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>190</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>190</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>10</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>400</width>
      <height>1</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>210</y>
      <width>400</width>
      <height>1</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>620</y>
      <width>400</width>
      <height>1</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>10</y>
      <width>1</width>
      <height>400</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>10</y>
      <width>1</width>
      <height>400</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>420</y>
      <width>1</width>
      <height>400</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="BezierWidget" name="bezierWidget">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>420</y>
      <width>400</width>
      <height>400</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>830</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Control Point</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>830</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove Control Point</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>830</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Wireframe</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>830</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Gouraud</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>830</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Phong</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>120</red>
          <green>120</green>
          <blue>120</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>TOP</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>10</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>120</red>
          <green>120</green>
          <blue>120</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>LEFT SIDE</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>420</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>120</red>
          <green>120</green>
          <blue>120</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>FRONT</string>
    </property>
   </widget>
  </widget>
  <action name="actionWireframe">
   <property name="text">
    <string>Wireframe</string>
   </property>
  </action>
  <action name="actionGouraud">
   <property name="text">
    <string>Gouraud</string>
   </property>
  </action>
  <action name="actionPhong">
   <property name="text">
    <string>Phong</string>
   </property>
  </action>
  <action name="actionAdd">
   <property name="text">
    <string>Add</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>Delete</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CPWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>cpwidget.h</header>
   <slots>
    <signal>ctrlPointMoved(QVector3D)</signal>
    <signal>ctrlPointSelected(int)</signal>
    <slot>moveCtrlPoint(QVector3D)</slot>
    <slot>selectCtrlPoint(int)</slot>
    <slot>addCtrlPoint()</slot>
    <slot>deleteCtrlPoint()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>BezierWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>bezierwidget.h</header>
   <slots>
    <slot>deleteCtrlPoint()</slot>
    <slot>addCtrlPoint()</slot>
    <slot>moveCtrlPoint(QVector3D)</slot>
    <slot>selectCtrlPoint(int)</slot>
    <slot>setWireframe()</slot>
    <slot>setGouraud()</slot>
    <slot>setPhong()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>topViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>sideViewWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel">
     <x>739</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>frontViewWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>frontViewWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>579</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>sideViewWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>409</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>590</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>topViewWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>453</y>
    </hint>
    <hint type="destinationlabel">
     <x>95</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>topViewWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>453</y>
    </hint>
    <hint type="destinationlabel">
     <x>64</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>topViewWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>543</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>topViewWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>542</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>sideViewWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>409</x>
     <y>453</y>
    </hint>
    <hint type="destinationlabel">
     <x>573</x>
     <y>312</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>sideViewWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>409</x>
     <y>453</y>
    </hint>
    <hint type="destinationlabel">
     <x>614</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>frontViewWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>652</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>frontViewWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>677</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>bezierWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>332</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>486</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>bezierWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>467</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>bezierWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>766</x>
     <y>389</y>
    </hint>
    <hint type="destinationlabel">
     <x>760</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>bezierWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>779</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>785</x>
     <y>509</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontViewWidget</sender>
   <signal>ctrlPointMoved(QVector3D)</signal>
   <receiver>bezierWidget</receiver>
   <slot>moveCtrlPoint(QVector3D)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>699</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>709</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontViewWidget</sender>
   <signal>ctrlPointSelected(int)</signal>
   <receiver>bezierWidget</receiver>
   <slot>selectCtrlPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>719</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>722</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>frontViewWidget</receiver>
   <slot>addCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>10</x>
     <y>841</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>780</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>bezierWidget</receiver>
   <slot>addCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>839</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>780</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>sideViewWidget</receiver>
   <slot>addCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>838</y>
    </hint>
    <hint type="destinationlabel">
     <x>521</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>topViewWidget</receiver>
   <slot>addCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>839</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>bezierWidget</receiver>
   <slot>deleteCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>841</y>
    </hint>
    <hint type="destinationlabel">
     <x>719</x>
     <y>686</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>frontViewWidget</receiver>
   <slot>deleteCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>194</x>
     <y>842</y>
    </hint>
    <hint type="destinationlabel">
     <x>278</x>
     <y>699</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>sideViewWidget</receiver>
   <slot>deleteCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>841</y>
    </hint>
    <hint type="destinationlabel">
     <x>695</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>topViewWidget</receiver>
   <slot>deleteCtrlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>836</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>362</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>bezierWidget</receiver>
   <slot>setWireframe()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>620</x>
     <y>843</y>
    </hint>
    <hint type="destinationlabel">
     <x>609</x>
     <y>732</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>bezierWidget</receiver>
   <slot>setGouraud()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>709</x>
     <y>841</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>778</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>bezierWidget</receiver>
   <slot>setPhong()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>789</x>
     <y>838</y>
    </hint>
    <hint type="destinationlabel">
     <x>782</x>
     <y>807</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
